server/src/Persistence.hs

```diff
@@ -173,12 +173,19 @@ recoverJournal pval = for_ (pvJournal pval) $ \journalHandle -> ExceptT $ fmap f
     failedRecoveryMsg err line = "Failed to recover journal entry "
       <> Text.pack (show line) <> ": " <> Text.pack err
 
--- | Read and decode the data file
-readData :: FilePath -> ExceptT String IO Store.Value
-readData filePath = ExceptT $ do
+-- | Read and decode the data file from disk
+readData :: FilePath -> Logger -> ExceptT String IO Store.Value
+readData filePath logger = ExceptT $ do
   eitherEncodedValue <- try $ withFile filePath ReadMode SBS.hGetContents
   case (eitherEncodedValue :: Either SomeException SBS.ByteString) of
-    Left exc -> pure $ Left $ "Failed to read the data from disk: " ++ show exc
+    Left _exc -> do
```

Comment:
This should check whether the exception was indeed that the file was not found, and only in that case it should attempt to create it.

Reply:
Good point, fixed.